### DISCUSSION TITLE
XML to ReqIF?

### DISCUSSION THREAD

**Discussion by vardaanti** [strictdoc-project/reqif#141] [Q&A]
Hi I am quite new to XML and ReqIF. I know that ReqIF is a type of XML file but from what I understand, it is formatted differently. Can XML be converted to this format using the unparser?

**Comment by stanislaw** [comment]
Hi! 

> Can XML be converted to this format using the unparser?

No, the function of the unparser is to parse the ReqIF in-memory representation (Python objects) to a .reqif file on a file system.

Could you share which problem are you trying to solve?

**Reply by vardaanti** [reply]
Yes, I am working on a cybersecurity project and this involves taking safety requirements from one tool and importing them into another. The problem we are running into is that the tool we are exporting from exports these requirements in XML and the tool we are trying to import into does so with ReqIF. I was hoping to be able to use this library to convert from XML to ReqIF to make this transfer possible. I noticed you mentioned that the deparser converts python elements into a .reqif file. I have been able to parse the XML file and turn some fields into python variables. Would I then be able to use the deparser to create a ReqIF file from these variables?

**Reply by stanislaw** [reply]
Hey @vardaanti . I apologize for not coming back to you right away. I managed to miss your message which normally happens very rarely because I read all of them. 

You definitely need to create the ReqIF Python objects from the content that you are parsing from XML. In fact, the job of a ReqIF parser itself is to parse ReqIF XML and construct the Python objects, so you could also follow what the parser does to do that, see https://github.com/strictdoc-project/reqif/blob/main/reqif/parser.py.

See also how a ReqIF file is created from the SDoc Python objects: https://github.com/strictdoc-project/strictdoc/blob/main/strictdoc/backend/reqif/p01_sdoc/sdoc_to_reqif_converter.py#L265

If this task is still actual to you, let me know if you could figure it out yourself or if I should create a tutorial that would show how to create a basic ReqIF structure from a bunch of sections/requirements with roughly the following structure:

Specification document
Section 1
  Requirement 1.1
  Requirement 1.2
Section 2
  Requirement 2.1
  Requirement 2.2

or something like that.

Now that you asked, I would create this tutorial in any case, the only question is how soon should I do that because there are some things piling up on the other tracks.
